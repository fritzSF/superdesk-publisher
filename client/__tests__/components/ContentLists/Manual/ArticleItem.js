import React from "react";
import ArticleItem from "../../../../components/ContentLists/Manual/ArticleItem";
import { render } from "@testing-library/react";

describe("ContentLists/Manual/ArticleItem", () => {
  const item = {
    content: {
      title: "test",
      route: {
        name: "route"
      },
      authors: [{ name: "author" }],
      published_at: "2017-07-26T13:43:33+00:00",
      article_statistics: {
        page_views_number: 10,
        internal_click_rate: 5,
        impressions_number: 15
      }
    }
  };

  it("renders correctly without extras", () => {
    const { container } = render(
      <ArticleItem
        item={item}
        openPreview={jest.fn()}
        previewItem={{}}
        index={1}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with extras", () => {
    const { container } = render(
      <ArticleItem
        item={item}
        openPreview={jest.fn()}
        previewItem={{}}
        showExtras={true}
        index={1}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
