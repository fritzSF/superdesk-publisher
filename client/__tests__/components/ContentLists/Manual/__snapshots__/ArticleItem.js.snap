// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLists/Manual/ArticleItem renders correctly with extras 1`] = `
<div
  class="sd-list-item sd-list-item--activated"
>
  <div
    class="sd-list-item__column"
  >
    <span
      class="sd-list-item__serial-number sp-listSerialNumberMinWidth"
    >
      2
    </span>
  </div>
  <div
    class="sd-list-item__column sd-list-item__column--grow"
  >
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item__text-strong"
      />
    </div>
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item--element-grow"
      >
        a few seconds ago
      </span>
      <span
        class="label label--success label--hollow"
      />
    </div>
  </div>
  <div
    class="sd-list-item__action-menu"
  >
    <button
      title="Remove"
    >
      <i
        class="icon-trash"
      />
    </button>
  </div>
</div>
`;

exports[`ContentLists/Manual/ArticleItem renders correctly without extras 1`] = `
<div
  class="sd-list-item sd-list-item--activated"
>
  <div
    class="sd-list-item__column sd-list-item__column--grow"
  >
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item__text-strong"
      />
    </div>
    <div
      class="sd-list-item__row"
    >
      <span
        class="sd-overflow-ellipsis sd-list-item--element-grow"
      >
        a few seconds ago
      </span>
      <span
        class="label label--success label--hollow"
      />
    </div>
  </div>
</div>
`;
